tests/test-war-smoke/pom.xml

```diff
@@ -228,35 +165,161 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-failsafe-plugin</artifactId>
-            <version>2.19.1</version>
-            <configuration>
-              <systemPropertyVariables>
-                <app.deploy.project>${app.deploy.project}</app.deploy.project>
-                <app.deploy.version>${maven.build.timestamp}</app.deploy.version>
-              </systemPropertyVariables>
-            </configuration>
             <executions>
               <execution>
+                <id>remote-integration-test</id>
                 <phase>integration-test</phase>
+                <goals>
+                  <goal>integration-test</goal>
+                  <goal>verify</goal>
+                </goals>
+                <configuration>
+                  <skipTests>false</skipTests>
+                  <systemPropertyVariables>
+                    <test.mode>remote</test.mode>
+                    <app.deploy.project>${app.deploy.project}</app.deploy.project>
+                    <app.deploy.version>${maven.build.timestamp}</app.deploy.version>
+                  </systemPropertyVariables>
+                </configuration>
               </execution>
             </executions>
           </plugin>
           <plugin>
-            <groupId>com.google.cloud.tools</groupId>
-            <artifactId>appengine-maven-plugin</artifactId>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
             <executions>
+              <!-- deploy the service -->
               <execution>
-                <id>deploy-integration-test</id>
+                <id>deploy-remote-service</id>
                 <phase>pre-integration-test</phase>
                 <goals>
-                  <goal>deploy</goal>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <workingDirectory>${project.build.directory}/appengine-staging</workingDirectory>
+                  <executable>gcloud</executable>
+                  <arguments>
+                    <argument>--quiet</argument>
+                    <argument>app</argument>
+                    <argument>deploy</argument>
+                    <argument>app.yaml</argument>
+                    <argument>--promote</argument>

```

Comment:
If there is an uncleaned version, it shouldn't affect us because the URLs we construct in the tests are version-specific.

```java
      // construct the remote uri
      StringBuilder uri = new StringBuilder();
      uri.append("https://");
      uri.append(version).append("-dot-");
      if (serviceId != null) {
        uri.append(serviceId).append("-dot-");
      }
      uri.append(projectId);
      uri.append(".appspot.com/");
```